From 1abd914b1a171ea89899699bbec3e2a901e61bac Mon Sep 17 00:00:00 2001
From: Lorenzo Puliti <lorenzo.ru.g@gmail.com>
Date: Mon, 19 Aug 2019 14:58:38 +0200
Subject: [PATCH] update-service: move supervise directories in tmpfs

Create log and service's supervise directories under /run/
tmpfs (they were in /var/ previously).
Also try to replace old supervise directories pointing to /var
with new one pointing to /run, when possible.
---
 debian/contrib/update-service | 17 +++++++++++++++--
 1 file changed, 15 insertions(+), 2 deletions(-)

diff --git a/debian/contrib/update-service b/debian/contrib/update-service
index 7e72501..bb83f2b 100644
--- a/debian/contrib/update-service
+++ b/debian/contrib/update-service
@@ -63,11 +63,24 @@ case "$opt" in
     if test "${svdir#/etc/}" != "$svdir"; then
       if test ! -h "$svdir"/supervise; then
         rm -rf "$svdir"/supervise
-        ln -s /var/lib/runit/supervise/"$sv" "$svdir"/supervise
+        ln -s /run/runit/supervise/"$sv" "$svdir"/supervise
+      else
+        #934500 force the switch of supervise into /run, keep untill bullseye +1
+        if [ $(readlink "$svdir"/supervise) != /run/runit/supervise/"$sv" ]; then
+            rm -rf "$svdir"/supervise
+            ln -s /run/runit/supervise/"$sv" "$svdir"/supervise
+        fi
       fi
       if test -d "$svdir"/log && test ! -h "$svdir"/log/supervise; then
         rm -rf "$svdir"/log/supervise
-        ln -s /var/lib/runit/log/supervise/"$sv" "$svdir"/log/supervise
+        ln -s /run/runit/supervise/"$sv".log "$svdir"/log/supervise
+      fi
+      #934500 force the switch of supervise into /run, keep untill bullseye +1
+      if [ -h "$svdir"/log/supervise ]; then
+        if [ $(readlink "$svdir"/log/supervise) != /run/runit/supervise/"$sv".log ]; then
+            rm -rf "$svdir"/log/supervise
+            ln -s /run/runit/supervise/"$sv".log "$svdir"/log/supervise
+        fi
       fi
     fi
     ln -s "$svdir" "$servicedir"/"$sv"
-- 
2.23.0

